wurst/systems/boards/ObserverBoard.wurst

```diff
@@ -0,0 +1,203 @@
+            let zoomStr = zoom.toString()
+            diag
+            .addButton(zoomStr, 0)
+            .onClick() btn ->
+                setZoom(zoom)
+                callback.call(zoomStr)
+
+        diag.display(whichPlayer, true)
+
+class ObserverBoard
+    static let boards = new HashMap<player, thistype>()
+
+    player observer
+    LinkedList<ObserverBoardControl> options
+    let board = CreateMultiboard()
+    var selected = 0
+    var minimized = false
+
+    construct(player observer, LinkedList<ObserverBoardControl> options)
+        this.observer = observer
+        this.options = options
+
+        boards.put(observer, this)
+
+        board
+            ..setColumnCount(2)
+            ..setRowCount(options.size())
+            ..display(observer, true)
+            ..setItemsWidth(0.1)
+
+        var row = 0
+        for option in options
+            board
+                .getItem(row, 0)
+                ..setValue(option.getName())
+                ..setValueColor(OPTION_NAME_COLOR)
+                ..setStyle(true, false)
+                ..setWidth(0.05)
+                ..release()
+
+            board
+                .getItem(row, 1)
+                ..setValue(option.getInitialValue())
+                ..setValueColor(OPTION_VALUE_COLOR)
+                ..setStyle(true, false)
+                ..release()
+
+            row += 1
+
+        doPeriodically(0.1) cb ->
+            board.setTitle(getGameTimersBoardTitle())
+            let isMinimized = board.isMinimized()
+            if minimized and not isMinimized
+                displayOptions()
+            minimized = isMinimized
+
+    function displayOptions()
+        let diag = DialogCreate()
+        for option in options
+            diag
+            .addButton(option.getName(), 0)
+            .onClick(btn -> begin
+                let row = options.indexOf(option)
+                option.onSelect(value -> begin
+                    board
+                        .getItem(row, 1)
+                        ..setValue(value)
+                        ..release()
+                end)
+            end)
+
+        diag
+            .addButton("Quit", 0)
+            .onClick(btn -> skip)
+
+        diag.display(observer, true)
+
+function getObservers() returns LinkedList<player>
+    let observers = new LinkedList<player>()
+    for i = 0 to bj_MAX_PLAYERS - 1
+        let current = players[i]
+        if current.isObserver() and current.isIngame()
+            observers.add(current)
+    return observers
+
+init
+    GameMode.onModeSelectionFinish() ->
+        let observers = getObservers()
+        for observer in observers
+            let options = asList<ObserverBoardControl>(
+                new ObserverBoardVisionControl(observer),
+                new ObserverBoardZoomControl(observer)
+            )
+            new ObserverBoard(observer, options)
+
+        destroy observers
+
+    registerPlayerEvent(EVENT_PLAYER_END_CINEMATIC) ->
```

Comment:
What is this supposed to trigger on?

Reply:
`Esc` button

Reply:
That gets fired even if there isn't a cinematic to end?

Reply:
Yes